test/unittest/dds/participant/ParticipantTests.cpp

```diff
@@ -977,6 +1026,1007 @@ TEST(ParticipantTests, SetListener)
+ * 4. Check that the previous operation returns an Inconsistent Policy error code
+ */
+TEST(ParticipantTests, CheckDomainParticipantQos)
+{
+    // Create the participant factory
+    DomainParticipantFactory* factory = DomainParticipantFactory::get_instance();
+
+    // Disable automatic entity enablement on the participant
+    {
+        DomainParticipantFactoryQos qos;
+        qos.entity_factory().autoenable_created_entities = false;
+
+        ASSERT_TRUE(factory->set_qos(qos) == ReturnCode_t::RETCODE_OK);
+    }
+
+    // Create the participant
+    DomainParticipant* participant =
+            DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+
+    // Get the participant qos
+    DomainParticipantQos qos;
+    ASSERT_TRUE(participant->get_qos(qos) == ReturnCode_t::RETCODE_OK);
+
+    // Change the user data
+    qos.user_data().set_max_size(5);
+    std::vector<eprosima::fastrtps::rtps::octet> my_data {0, 1, 2, 3, 4};
+    qos.user_data().setValue(my_data);
+    ASSERT_TRUE(participant->set_qos(qos) == ReturnCode_t::RETCODE_OK);
+
+    // Change the ParticipantResourceLimitsQos to a maximum user data value less than the current user data size
+    // This should return an Inconsistent Policy error code
+    qos.allocation().data_limits.max_user_data = 1;
+    ASSERT_EQ(qos.allocation().data_limits.max_user_data, 1ul);
+    ASSERT_TRUE(participant->set_qos(qos) == ReturnCode_t::RETCODE_INCONSISTENT_POLICY);
+
+    // Enable the participant
+    EXPECT_EQ(ReturnCode_t::RETCODE_OK, participant->enable());
+
+    // Remove the participant
+    ASSERT_TRUE(DomainParticipantFactory::get_instance()->delete_participant(participant) == ReturnCode_t::RETCODE_OK);
+}
+
+/*
+ * This test checks the cases in which the allocation QoS is modified.
+ * 1. Check that the qos is modified if the participant is not enabled.
+ * 2. Check that the qos is not changed and it generates an error code if the participant is already enabled.
+ */
+TEST(ParticipantTests, ChangeAllocationDomainParticipantQos)
+{
+    DomainParticipantFactory* factory = DomainParticipantFactory::get_instance();
+    DomainParticipantFactoryQos pfqos;
+    pfqos.entity_factory().autoenable_created_entities = false;
+    ASSERT_EQ(factory->set_qos(pfqos), ReturnCode_t::RETCODE_OK);
+
+    DomainParticipant* participant =
+            DomainParticipantFactory::get_instance()->create_participant(0, PARTICIPANT_QOS_DEFAULT);
+    ASSERT_FALSE(participant->is_enabled());
+    DomainParticipantQos qos;
+    participant->get_qos(qos);
+
+    ASSERT_EQ(qos, PARTICIPANT_QOS_DEFAULT);
+
+    qos.allocation().data_limits.max_properties = 10;
+    ASSERT_EQ(participant->set_qos(qos), ReturnCode_t::RETCODE_OK);
+    DomainParticipantQos pqos;
+    participant->get_qos(pqos);
+
+    ASSERT_FALSE(pqos == PARTICIPANT_QOS_DEFAULT);
+    ASSERT_EQ(qos, pqos);
+    ASSERT_EQ(pqos.allocation().data_limits.max_properties, 10);
```

Comment:
```suggestion
    ASSERT_EQ(pqos.allocation().data_limits.max_properties, 10ul);
```
Warning in MAC Jenkins gtest version.